tests/unit/tuner/torch/test_miner.py

```diff
@@ -75,3 +95,60 @@ def test_triplet_miner_given_insufficient_inputs(
     labels = labels[:cut_index]
     rv = list(siamese_miner.mine(embeddings, labels))
     assert len(rv) == 0
+
+
+def test_siamese_session_miner(embeddings, session_labels, siamese_session_miner):
+    rv = siamese_session_miner.mine(embeddings, session_labels)
+    assert (
+        len(rv) == 8
+    )  # 1 pair with session 1, 1 pair with session 2, 6 pairs with session 3
+    for item in rv:
+        idx_left, idx_right, label = item
+        # find corresponded label idx
+        label_left = session_labels[idx_left]
+        label_right = session_labels[idx_right]
+        if label_left == label_right:
+            expected_label = 1
+        else:
+            expected_label = -1
+        assert label == expected_label
```

Comment:
Here I would prefer if you explicitly list the tuples you expect to be produced (since it's only 8 of them, it's feasible) or produce them by a "reliable" method.

The way test is written now does not pin down the result enough in my opinion